File: backend-intermediate-cs/OrderManagement/Models/Order.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace OrderManagement.Models {
    [Table("orders")]
    public class Order
    {
        [Key]
        [Column("order_number")]
        public int OrderNumber { get; set; }

        [Column("order_date")]
        [DataType(DataType.Date)]
        public DateOnly OrderDate { get; set; } = DateOnly.FromDateTime(DateTime.Now);
        
        [Column("total")]
        public double Total { get; set; }

        [Required]
        [Column("customer_id")]
        public int CustomerId { get; set; }

        // Navigation properties
        public Customer? Customer { get; set; }
        public ICollection<OrderDetail> Details { get; set; } = new List<OrderDetail>();
    }
}

File: backend-intermediate-java/OrderManagementJava/src/main/java/org/orders/model/Order.java
package org.orders.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.Getter;
import lombok.Setter;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "orders")
@Getter
@Setter
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "order_number")
    private Integer orderNumber;

    @Column(name = "order_date")
    private LocalDate orderDate = LocalDate.now();

    @Column(name = "total")
    private Double total;

    @NotNull
    @Column(name = "customer_id", nullable = false)
    private Integer customerId;

    // Navigation properties
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "customer_id", insertable = false, updatable = false)
    private Customer customer;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
    private List<OrderDetail> details = new ArrayList<>();
}
